mula/scheduler/server/server.py

```diff
@@ -409,26 +409,26 @@ def push_queue(self, queue_id: str, item: models.PrioritizedItem) -> Any:
                 p_item.data = models.NormalizerTask(**p_item.data).dict()
         except Exception as exc:
             raise fastapi.HTTPException(
-                status_code=400,
+                status_code=500,
                 detail=str(exc),
             ) from exc
 
         try:
             s.push_item_to_queue(p_item)
-        except queues.QueueFullError as exc_full:
-            raise fastapi.HTTPException(
-                status_code=400,
-                detail="queue is full",
-            ) from exc_full
         except ValueError as exc_value:
             raise fastapi.HTTPException(
                 status_code=400,
                 detail="invalid item",
             ) from exc_value
+        except queues.QueueFullError as exc_full:
+            raise fastapi.HTTPException(
+                status_code=429,
```

Comment:
Although it would be convenient to have this one, strictly speaking, 429s are used for rate-limiting requests. They can be applied to single endpoints or on a middleware layer (that e.g. rate limits based on a token or resource id). Having a full queue is more of a resource problem. I'll leave it up to you.

See also point below about status constants.

Reply:
Yeah, I noticed as well that it is used specifically for rate-limiting requests, it is a bit of a toss-up. My consideration was as follows: The HTTP 409 status code indicates that the request could not be completed due to a conflict with the current state of the target resource. In this case, the conflict arises from the fact that the item we're trying to push onto the queue is already present, and therefore the operation cannot be completed as requested. In that regard, it could be considered a rate-limiting problem, and we could even relay a rate-limiting header to the client who wants to wait for a set amount of time.

Reply:
In that case, it would indeed be more justified if we add a rate-limiting header